src/library/commissioning_session.cpp

```diff
@@ -227,12 +200,79 @@ Error CommissioningSession::SendJoinFinResponse(const coap::Request &aJoinFinReq
     coap::Response joinFin{coap::Type::kAcknowledgment, coap::Code::kChanged};
     SuccessOrExit(error = AppendTlv(joinFin, {tlv::Type::kState, aAccept ? tlv::kStateAccept : tlv::kStateReject}));
 
+    joinFin.SetSubType(MessageSubType::kJoinFinResponse);
     SuccessOrExit(error = mCoap.SendResponse(aJoinFinReq, joinFin));
 
 exit:
     return error;
 }
 
+CommissioningSession::RelaySocket::RelaySocket(CommissioningSession &aCommissioningSession,
+                                               const Address &       aPeerAddr,
+                                               uint16_t              aPeerPort,
+                                               const Address &       aLocalAddr,
+                                               uint16_t              aLocalPort)
+    : Socket(aCommissioningSession.mCommImpl.GetEventBase())
+    , mCommissioningSession(aCommissioningSession)
+    , mPeerAddr(aPeerAddr)
+    , mPeerPort(aPeerPort)
+    , mLocalAddr(aLocalAddr)
+    , mLocalPort(aLocalPort)
+{
+    mIsConnected = true;
+
+    int fail = event_assign(&mEvent, mEventBase, -1, EV_PERSIST | EV_READ | EV_WRITE | EV_ET, HandleEvent, this);
+    ASSERT(fail == 0);
+    ASSERT((fail = event_add(&mEvent, nullptr)) == 0);
```

Comment:
Avoid including logic in assert statement?

Reply:
The `ASSERT` here is implemented as `abort()` if the condition is not satisfied (it allows side-effect). The name may be not suitable, I have been thinking rename it to `Check(...)` or `VerifyOrDie(...)`. What do you think?

Reply:
Of the options, I think `VerifyOrDie()` is more descriptive and better than `ASSERT`.

Reply:
okay, I will do renaming it in separate PR.